Scenario crash when UPDATE OR ROLLBACK to the on Android

when update or rollback is done in android application, the application crashes. The process is completed successfully, the update is done, but the application crashes when it will reload. 
it makes no difference whether it is force or not. 
No problem on iOS. 

Device samsung s24 ultra
react native : 0.76.5
android version 14

logcat logs 

 java.lang.RuntimeException: Could not invoke HotUpdater.updateBundle
                                                                                                    	at com.facebook.react.bridge.JavaMethodWrapper.invoke(Unknown Source:188)
                                                                                                    	at com.facebook.react.bridge.JavaModuleWrapper.invoke(Unknown Source:19)
                                                                                                    	at com.facebook.jni.NativeRunnable.run(Native Method)
                                                                                                    	at android.os.Handler.handleCallback(Handler.java:958)
                                                                                                    	at android.os.Handler.dispatchMessage(Handler.java:99)
                                                                                                    	at com.facebook.react.bridge.queue.MessageQueueThreadHandler.dispatchMessage(Unknown Source:0)
                                                                                                    	at android.os.Looper.loopOnce(Looper.java:230)
                                                                                                    	at android.os.Looper.loop(Looper.java:319)
                                                                                                    	at com.facebook.react.bridge.queue.MessageQueueThreadImpl.lambda$startNewBackgroundThread$2(Unknown Source:35)
                                                                                                    	at com.facebook.react.bridge.queue.MessageQueueThreadImpl.b(Unknown Source:0)
                                                                                                    	at com.facebook.react.bridge.queue.b.run(Unknown Source:2)
                                                                                                    	at java.lang.Thread.run(Thread.java:1012)
                                                                                                    Caused by: java.lang.reflect.InvocationTargetException
                                                                                                    	at java.lang.reflect.Method.invoke(Native Method)
                                                                                                    	at com.facebook.react.bridge.JavaMethodWrapper.invoke(Unknown Source:150)
                                                                                                    	at com.facebook.react.bridge.JavaModuleWrapper.invoke(Unknown Source:19) 
                                                                                                    	at com.facebook.jni.NativeRunnable.run(Native Method) 
                                                                                                    	at android.os.Handler.handleCallback(Handler.java:958) 
                                                                                                    	at android.os.Handler.dispatchMessage(Handler.java:99) 
                                                                                                    	at com.facebook.react.bridge.queue.MessageQueueThreadHandler.dispatchMessage(Unknown Source:0) 
                                                                                                    	at android.os.Looper.loopOnce(Looper.java:230) 
                                                                                                    	at android.os.Looper.loop(Looper.java:319) 
                                                                                                    	at com.facebook.react.bridge.queue.MessageQueueThreadImpl.lambda$startNewBackgroundThread$2(Unknown Source:35) 
                                                                                                    	at com.facebook.react.bridge.queue.MessageQueueThreadImpl.b(Unknown Source:0) 
                                                                                                    	at com.facebook.react.bridge.queue.b.run(Unknown Source:2) 
                                                                                                    	at java.lang.Thread.run(Thread.java:1012) 
                                                                                                    Caused by: java.lang.IllegalAccessException: Could not setJSBundle
                                                                                                    	at Tc.d.d(Unknown Source:84)
                                                                                                    	at com.hotupdater.a$a.i(Unknown Source:47)
                                                                                                    	at com.hotupdater.a$a.k(Unknown Source:321)
                                                                                                    	at com.hotupdater.HotUpdaterModule.updateBundle(Unknown Source:19)
                                                                                                    	at java.lang.reflect.Method.invoke(Native Method) 
                                                                                                    	at com.facebook.react.bridge.JavaMethodWrapper.invoke(Unknown Source:150) 
                                                                                                    	at com.facebook.react.bridge.JavaModuleWrapper.invoke(Unknown Source:19) 
                                                                                                    	at com.facebook.jni.NativeRunnable.run(Native Method) 
                                                                                                    	at android.os.Handler.handleCallback(Handler.java:958) 
                                                                                                    	at android.os.Handler.dispatchMessage(Handler.java:99) 
                                                                                                    	at com.facebook.react.bridge.queue.MessageQueueThreadHandler.dispatchMessage(Unknown Source:0) 
                                                                                                    	at android.os.Looper.loopOnce(Looper.java:230) 
                                                                                                    	at android.os.Looper.loop(Looper.java:319) 
                                                                                                    	at com.facebook.react.bridge.queue.MessageQueueThreadImpl.lambda$startNewBackgroundThread$2(Unknown Source:35) 
                                                                                                    	at com.facebook.react.bridge.queue.MessageQueueThreadImpl.b(Unknown Source:0) 
                                                                                                    	at com.facebook.react.bridge.queue.b.run(Unknown Source:2) 
                                                                                                    	at java.lang.Thread.run(Thread.java:1012) 

--- a/packages/react-native/android/build.gradle
+++ b/packages/react-native/android/build.gradle
@@ -65,7 +65,7 @@ android {
     minSdkVersion getExtOrIntegerDefault("minSdkVersion")
     targetSdkVersion getExtOrIntegerDefault("targetSdkVersion")
     buildConfigField "boolean", "IS_NEW_ARCHITECTURE_ENABLED", isNewArchitectureEnabled().toString()
-
+    consumerProguardFiles 'proguard-rules.pro'
   }
 
   buildFeatures {

--- a/packages/react-native/android/proguard-rules.pro
+++ b/packages/react-native/android/proguard-rules.pro
@@ -1,0 +1,25 @@
+# Add project specific ProGuard rules here.
+# By default, the flags in this file are appended to flags specified
+# in /usr/local/Cellar/android-sdk/24.3.3/tools/proguard/proguard-android.txt
+# You can edit the include path and order by changing the proguardFiles
+# directive in build.gradle.
+#
+# For more details, see
+#   http://developer.android.com/guide/developing/tools/proguard.html
+
+# Add any project specific keep options here:
+
+# If your project uses WebView with JS, uncomment the following
+# and specify the fully qualified class name to the JavaScript interface
+# class:
+#-keepclassmembers class fqcn.of.javascript.interface.for.webview {
+#   public *;
+#}
+
+# Invoked via reflection, when setting js bundle.
+-keepclassmembers class com.facebook.react.ReactInstanceManager {
+    private final ** mBundleLoader;
+}
+
+# Can't find referenced class org.bouncycastle.**
+-dontwarn com.nimbusds.jose.**

--- a/packages/react-native/android/proguard-rules.pro
+++ b/packages/react-native/android/proguard-rules.pro
@@ -20,6 +20,3 @@
 -keepclassmembers class com.facebook.react.ReactInstanceManager {
     private final ** mBundleLoader;
 }
-
-# Can't find referenced class org.bouncycastle.**
--dontwarn com.nimbusds.jose.**

